src/api_manager/service_control/aggregated.cc

```diff
@@ -371,8 +371,9 @@ void Aggregated::Check(
           *response, service_control_proto_.service_name(), &response_info);
       on_done(status, response_info);
     } else {
-      // If network_fail_open is true, it is always OK to proceed.
-      if (network_fail_open_) {
+      // Only Code::UNAVAILABLE error is network failure.
+      // If network_fail_open is true, it is OK to proceed
+      if (network_fail_open_ && status.error_code() == Code::UNAVAILABLE) {
```

Comment:
Would be better to limit to all [5XX](https://www.websitepulse.com/kb/5xx_http_status_codes) errors not only service unavaliable? 
Also, please change the title to be
"Make fail open only on server side problem"

Reply:
Done